DelvEdit/src/com/interrupt/dungeoneer/editor/ui/menu/generator/GeneratorInfo.java

```diff
@@ -0,0 +1,125 @@
+        for (String mod : mods) {
+            refreshGenerator(mod);
+            refreshData(mod);
+        }
+
+        themes.sort();
+    }
+
+    private void refreshGenerator(String mod) {
+        FileHandle parent = Game.getInternal(mod + "/generator");
+        if (!parent.exists() || !parent.isDirectory()) {
+            return;
+        }
+
+        FileHandle[] children = parent.list(new FileFilter() {
+            @Override
+            public boolean accept(File file) {
+                return file.isDirectory();
+            }
+        });
+
+        for (FileHandle child : children) {
+            // Check for theme definition.
+            FileHandle info = Game.getInternal(mod + "/generator/" + child.name() + "/info.dat");
+            if (info.exists()) {
+                String theme = child.name().toUpperCase();
+
+                if (!themes.contains(theme, false)) {
+                    themes.add(theme);
+                }
+            }
+
+            // Check for room/level definition.
+            FileHandle section = Game.getInternal(mod + "/generator/" + child.name() + "/section.dat");
+            if (section.exists()) {
+                SectionDefinition sectionDefinition = Game.fromJson(SectionDefinition.class, section);
+
+                if (!sectionDefinitions.contains(sectionDefinition, false)) {
+                    sectionDefinitions.add(sectionDefinition);
+                }
+            }
+        }
+    }
+
+    private void refreshData(String mod) {
+        FileHandle parent = Game.getInternal(mod + "/data/room-builders");
+        if (!parent.exists() || !parent.isDirectory()) {
+            return;
+        }
+
+        FileHandle[] children = parent.list(new FileFilter() {
+            @Override
+            public boolean accept(File file) {
+                return !file.isDirectory();
+            }
+        });
+
+        for (FileHandle child : children) {
+            // Check for room builders.
+            String builder = child.nameWithoutExtension().toUpperCase();
+
+            if (!builders.contains(builder, false)) {
+                builders.add(builder);
+            }
+        }
+    }
+
+    public Array<String> getThemes() {
+        return themes;
+    }
+
+    public Array<SectionDefinition> getSectionDefinitions() {
+        Array<SectionDefinition> sortedSectionDefinitions = sectionDefinitions;
+        sortedSectionDefinitions.sort(new Comparator<SectionDefinition>() {
+            @Override
+            public int compare(SectionDefinition o1, SectionDefinition o2) {
+                return Integer.signum(o1.sortOrder - o2.sortOrder);
+            }
+        });
+
+        return sortedSectionDefinitions;
+    }
+
+    public boolean isLevelTemplateValid(Level template) {
+        if (!themes.contains(template.theme, false)) {
+            return false;
+        }
+
+        if (!builders.contains(template.roomGeneratorType, false)) {
```

Comment:
This and the below can be simplified to:
```java
return builders.contains(template.roomGeneratorType, false);
```

Reply:
Fixed.